dags/data_quality_checks_dag.py

```diff
@@ -0,0 +1,173 @@
+    start_date=datetime(2021, 1, 1),
+    schedule_interval=None,
+    catchup=False,
+    on_failure_callback=slack_failure_notification
+) as dag:
+    with TaskGroup(group_id="upload_local_files") as upload:
+        for file in os.listdir(FILE_DIR):
+            LocalFilesystemToS3Operator(
+                task_id=f"upload_{file}",
+                filename=f"{FILE_DIR}/{file}",
+                dest_key=f"incoming-data/{file}",
+                dest_bucket=S3_BUCKET,
+                aws_conn_id="aws_default",
+                replace=True,
+            )
+
+    import_stmt = get_import_statements(S3_BUCKET, "incoming")
+    upload >> import_stmt
+
+    import_data = PostgresOperator.partial(
+        task_id="import_data_to_cratedb",
+        postgres_conn_id="cratedb_connection"
+    ).expand(sql=import_stmt)
+
+    refresh = PostgresOperator(
+        task_id="refresh_table",
+        postgres_conn_id="cratedb_connection",
+        sql="""
+                REFRESH TABLE {{params.temp_table}};   
+            """,
+        params={
+                "table": TEMP_TABLE,
+        }
+    )
+    import_stmt >> import_data >> refresh
+
+    with TaskGroup(group_id="home_data_checks") as checks:
+        column_checks = SQLColumnCheckOperator.partial(
+            task_id="home_data_column_check",
+            conn_id="cratedb_connection",
+            table=TEMP_TABLE,
+        ).expand(column_mapping=COL_CHECKS)
+
+        # Table Check Operator requires Airflow 2.3.4 release
+        table_checks = SQLTableCheckOperator.partial(
+            task_id="home_data_table_check",
+            conn_id="cratedb_connection",
+            table=TEMP_TABLE,
+        ).expand(checks=TABLE_CHECKS)
+
+    move_data = PostgresOperator(
+        task_id="move_to_table",
+        postgres_conn_id="cratedb_connection",
+        sql="""
+                INSERT INTO {{params.table}} SELECT * FROM {{params.temp_table}};   
+            """,
+        params={
+                "table": TABLE,
+                "temp_table": TEMP_TABLE
+        }
+    )
+
+    delete_data = PostgresOperator(
+        task_id="delete_from_temp_table",
+        postgres_conn_id="cratedb_connection",
+        sql="""
+                DELETE FROM {{params.temp_table}};   
+            """,
+        params={
+                "temp_table": TEMP_TABLE
+        },
+        trigger_rule='all_done'
+    )
+
+    refresh >>  checks >> move_data >> delete_data
+
+    with TaskGroup(group_id="move_incoming_files") as processed:
+        for file in os.listdir(FILE_DIR):
+            S3CopyObjectOperator(
+                task_id=f"move_{file}",
+                aws_conn_id='aws_default',
+                source_bucket_name=S3_BUCKET,
+                source_bucket_key=f"incoming-data/{file}",
+                dest_bucket_name=S3_BUCKET,
+                dest_bucket_key=f"processed-data/{file}",
+            )
+
+    s3_files = get_files_from_s3(S3_BUCKET, "incoming")
```

Comment:
Should have this higher up and let the `get_import_statements()` have it as a dependency as well!

Reply:
Done!